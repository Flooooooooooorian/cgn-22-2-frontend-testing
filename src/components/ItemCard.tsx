import {Item} from "../Item";

type ItemCardProps = {
    item: Item
}

export default function ItemCard(props: ItemCardProps) {

    return (
        <>
            <p>
                {props.item.name}
            </p>
        </>
    )
}
