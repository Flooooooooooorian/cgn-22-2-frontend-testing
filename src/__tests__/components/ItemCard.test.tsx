import {Item} from "../../Item";
import {render} from "@testing-library/react";
import ItemCard from "../../components/ItemCard";


test("ItemCard test", () => {
    //GIVEN
    const testItem: Item = {id: "testId", name: "testName"};

    //WHEN
    const view = render(<ItemCard item={testItem} />)

    //THEN
    expect(view.asFragment()).toMatchSnapshot();
})
