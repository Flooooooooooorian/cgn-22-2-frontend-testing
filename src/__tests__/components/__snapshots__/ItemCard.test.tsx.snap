// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemCard test 1`] = `
<DocumentFragment>
  <p>
    testName
  </p>
</DocumentFragment>
`;
